instrumentation/opentelemetry-instrumentation-asyncpg/src/opentelemetry/instrumentation/asyncpg/__init__.py

```diff
@@ -105,16 +118,19 @@ async def _do_execute(self, func, instance, args, kwargs):
         tracer = getattr(asyncpg, _APPLIED)
 
         exception = None
-
-        with tracer.start_as_current_span(
-            "postgresql", kind=SpanKind.CLIENT
-        ) as span:
+        span_attributes = _hydrate_span_from_args(
+            instance, args[0], args[1:] if self.capture_parameters else None
+        )
+        name = ""
+        if args[0]:
+            name = args[0]
+        elif span_attributes.get("db.name"):
```

Comment:
Nit: Can you just use `getattr(params, "database", None)` to get the `db.name` instead of calling `_hydrate_span_from_args` before the span is created? Ideally, we don't want to call any unnecessary functions related to span attributes if the span is not recording, which is why originally hydrating the span occurs within the `is_recording()` check.

Reply:
Good point. I will change.